### DISCUSSION TITLE
Best practice when navigating to an activity containing children fragments

### DISCUSSION THREAD

**Discussion by DamienDoumer** [MvvmCross/MvvmCross#4063] [General]
I have a little question please. I might be using the mvvmcross framework badly.
When navigating from a normal activity to an activity with multiple fragments. Should I call the navigationtion service and pass to it the type of the mainactivity's viewmodel (the activity containing the fragments) or I should use the type of the child fragment's viewmodel to which I want to navigate ?

**Comment by Cheesebaron** [comment]
Unclear what you are asking. Do you have any examples that would help explain?

**Comment by Hackmodford** [comment]
I think you could do this two ways.

await navigation to the Activities viewModel, then call the navigation to each fragments viewmodels.

Or navigate to one of the fragments viewModels, then await navigation to the other fragment viewmodels.

**Comment by qwertylolman** [comment]
Fragment presenter attribute contains property called like ActivityViewModelType to configure the host activity.
After configuring this property you can navigate right to the fragment's viewmodel
